src/main.rs

```diff
@@ -384,11 +384,26 @@ fn show_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
     // first, get us an iterator over all the commits
     // NOTE: "issue" is a required parameter
     let issue = try_or_1!(Oid::from_str(matches.value_of("issue").unwrap()));
-    let graph = try_or_1!(repo.get_issue_revwalk(issue)).abort_on_err()
+    let commits = try_or_1!(repo.get_issue_revwalk(issue)).abort_on_err()
         .map(|oid| repo.find_commit(oid)).abort_on_err()
         .into_tree_graph();
 
-    // TODO: reverse the iterator if necessary (chronological vs. "timeline")
+    // Transform the simple graph element line into an iterator we may use for
+    // zipping with the message, reversing the order if necessary
+    let graph : Box<Iterator<Item=(CommitTreeGraphLines, Commit)>> = if matches.is_present("tree") {
+        // We want the commits in chronological order
+        let mut buf : Vec<(TreeGraphElemLine, Commit)> = commits.collect();
+        buf.reverse();
```

Comment:
I'll look into it.
I already considered something similar, but it would have meant using a `Box<Iterator>`. And if I have the choice between a boxed iterator over possibly pre-collected values and a straight `Vec`, I prefer the `Vec`.